### PR TITLE
handles instances where valid is false in the reponse with tests

### PR DESCRIPTION
**Pull request by adamzinder**

### CODE REVIEW

File: lib/workos/types/verify_factor_struct.rb [hunk 1]
```diff
@@ -7,9 +7,9 @@ module Types
...
     class VerifyFactorStruct < T::Struct
       const :challenge, T.nilable(T::Hash[Symbol, Object])
-      const :valid, T.nilable(TrueClass)
+      const :valid, T.any(TrueClass, FalseClass)
       const :code, T.nilable(String)
       const :message, T.nilable(String)
     end
   end
-end
+end
```
Comment (robframpton) [inline lib/workos/types/verify_factor_struct.rb:15]: We'll need to add this newline back to remove the Robocop warning that's failing CI.

**Review by robframpton — APPROVED** [review]

File: lib/workos/types/verify_factor_struct.rb (hunk 1 above)
Comment (adamzinder) [reply lib/workos/types/verify_factor_struct.rb]: done!

File: lib/workos/types/verify_factor_struct.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ module Types
     # for the Factor class
     class VerifyFactorStruct < T::Struct
       const :challenge, T.nilable(T::Hash[Symbol, Object])
-      const :valid, T.nilable(TrueClass)
+      const :valid, T.any(TrueClass, FalseClass)
```
Comment (marktran) [inline lib/workos/types/verify_factor_struct.rb:10]: Does `T::Boolean` work here @adamzinder @robframpton?

File: spec/lib/workos/mfa_spec.rb [hunk 3]
```diff
@@ -135,6 +135,17 @@
         end
       end
     end
+    context 'verify generic otp invalid response' do
```
Comment (marktran) [inline spec/lib/workos/mfa_spec.rb:138]: Can we add newlines around all of these `context` blocks, @adamzinder?

**Pull request merged**